src/main/java/org/sonar/scm/git/blame/FileBlamer.java

```diff
@@ -79,6 +91,22 @@ public void initialize(ObjectReader objectReader, GraphNode commit) {
     fileTreeComparator.initialize(objectReader);
   }
 
+  private void initializeForLargeFileSet(ObjectReader objectReader, GraphNode commit) {
+
+    Set<String> filesToCheck = commit.getAllFiles().stream().map(FileCandidate::getPath).collect(Collectors.toSet());
+    try {
+      Map<String, Integer> fileSizes = fileReader.getFileSizes(filesToCheck);
+      for (FileCandidate fileCandidate : commit.getAllFiles()) {
+        int fileSize = fileSizes.get(fileCandidate.getPath());
```

Comment:
isn't there a risk that we didn't find the file during the tree walk and this ends up throwing a NPE?

Reply:
Correct. I will add a check and throw and `IllegalStateException` as in the other alorithm using `loadText`

Reply:
Done